scripts/2_match_households_and_individuals.py

```diff
@@ -0,0 +1,1298 @@
+import itertools
+import os
+import pickle as pkl
+
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+from IPython.display import display
+from tqdm import trange
+
+from acbm.matching import match_categorical, match_individuals
+from acbm.preprocessing import (
+    count_per_group,
+    # nts_filter_by_region,
+    nts_filter_by_year,
+    num_adult_child_hh,
+    transform_by_group,
+    truncate_values,
+)
+
+pd.set_option("display.max_columns", None)
+
+
+def get_interim_path(file_name: str, path: str = "../data/interim/matching/") -> str:
+    os.makedirs(path, exist_ok=True)
+    return f"{path}/{file_name}"
+
+
+# ## Step 1: Load in the datasets
+
+# ### SPC
+
+# useful variables
+region = "west-yorkshire"
+
+# Read in the spc data (parquet format)
+spc = pd.read_parquet("../data/external/spc_output/" + region + "_people_hh.parquet")
+
+# select columns
+spc = spc[
+    [
+        "id",
+        "household",
+        "pid_hs",
+        "msoa11cd",
+        "oa11cd",
+        "members",
+        "sic1d2007",
+        "sic2d2007",
+        "pwkstat",
+        "salary_yearly",
+        "salary_hourly",
+        "hid",
+        "accommodation_type",
+        "communal_type",
+        "num_rooms",
+        "central_heat",
+        "tenure",
+        "num_cars",
+        "sex",
+        "age_years",
+        "ethnicity",
+        "nssec8",
+    ]
+]
+
+
+# temporary reduction of the dataset for quick analysis
+# TODO: check if this should be present?
+spc = spc.head(15000)
```

Comment:
Perhaps can be an input parameter for the script to determine if run on the whole population?